File: frontend-user/src/utils/logger.ts
const isProd = import.meta.env.PROD

export const logger = {
  debug(...args: unknown[]) {
    if (!isProd) console.debug('[gojira]', ...args)
  },
  info(...args: unknown[]) {
    if (!isProd) console.info('[gojira]', ...args)
  },
  warn(...args: unknown[]) {
    console.warn('[gojira]', ...args)
  },
  error(...args: unknown[]) {
    console.error('[gojira]', ...args)
  },
}

File: backend/internal/logger/logger.go
package logger

import (
	"log/slog"
	"os"
	"strings"
)

// New builds the process-wide slog JSON logger. Production (and any non-debug
// level) drops debug records. Callers must not use fmt.Println.
func New(level string) *slog.Logger {
	var lvl slog.Level
	switch strings.ToLower(strings.TrimSpace(level)) {
	case "debug":
		lvl = slog.LevelDebug
	case "warn", "warning":
		lvl = slog.LevelWarn
	case "error":
		lvl = slog.LevelError
	default:
		lvl = slog.LevelInfo
	}
	h := slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
		Level: lvl,
		ReplaceAttr: func(_ []string, a slog.Attr) slog.Attr {
			if a.Key == slog.TimeKey {
				return a
			}
			return a
		},
	})
	log := slog.New(h).With("service", "gojira")
	slog.SetDefault(log)
	return log
}
